src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -333,7 +337,7 @@ public Object call() throws Exception
                 }
             }
             FBUtilities.waitOnFutures(futures);
-            assert compacting.originals().isEmpty();
+            assert (compacting.originals().size() == originalCount - tableCount);
```

Comment:
this assert was incorrect.

There is no unit test for 'nodetool relocatesstables'  but if there was, this assert would break there too.  Any operation that reduces the number of SSTables returned from `operation.filterSSTables` would break.  I assume the assert came from a time prior, when all operations were supposed to process every file.